common/djangoapps/entitlements/models.py

```diff
@@ -404,6 +405,32 @@ def check_for_existing_entitlement_and_enroll(cls, user, course_run_key):
             return cls.enroll_user_and_fulfill_entitlement(entitlement, course_run_key)
         return False
 
+    @classmethod
+    def unenroll_entitlement(cls, course_enrollment, skip_refund):
+        """
+        Un-enroll the user from entitlement and refund.
+        """
+        course_uuid = get_course_uuid_for_course(course_enrollment.course_id)
+        course_entitlement = cls.get_entitlement_if_active(course_enrollment.user, course_uuid)
+        if course_entitlement:
+            course_entitlement.set_enrollment(None)
+            if not skip_refund and course_entitlement.is_entitlement_refundable():
+                course_entitlement.refund()
+
+    def refund(self):
+        """
+        Initiate refund process for the entitlement.
+        """
+        refund_successful = refund_entitlement(course_entitlement=self)
```

Comment:
I think we should add more checks to this code:
1. We can define a refund window for entitlement products with the help of Marco
2. We need to check if user have not earned a certificate for course enrollment for any paid tract. 
and not refund in those case and refund only if those cases are false. 

User can get refund after earning certificate.

Reply:
@Ayub-Khan We are already have those checks here: https://github.com/edx/edx-platform/blob/e28aa4657f9e10177b40d6c59df900d5fddda995/common/djangoapps/entitlements/models.py#L417

and here:
https://github.com/edx/edx-platform/blob/2f12e0b4c47e99aaaf900b5288d33f902aba791d/common/djangoapps/entitlements/api/v1/views.py#L475